File: address-validation-us-3-dot-net-examples/ValidateCityStateZipRestSdkExample.cs
using address_validation_us_3_dot_net.REST;

namespace address_validation_us_3_dot_net_examples
{
    internal static class ValidateCityStateZipRestSdkExample
    {
        public static void Go(string LicenseKey, bool IsLive)
        { 
            Console.WriteLine("\r\n---------------------------------------------------------");
            Console.WriteLine("Address Validation US 3 - ValidateCityStateZip - REST SDK");
            Console.WriteLine("---------------------------------------------------------");

            ValidateCityStateZipClient.ValidateCityStateZipInput validateCityStateZipInput = new(
                "Santa Barbara",
                "CA",
                "93101",
                LicenseKey,
                IsLive
            );
            Console.WriteLine("\r\n* Input *\r\n");

            Console.WriteLine($"City       : {validateCityStateZipInput.City}");
            Console.WriteLine($"State      : {validateCityStateZipInput.State}");
            Console.WriteLine($"ZIP Code   : {validateCityStateZipInput.Zip}");
            Console.WriteLine($"Is Live    : {validateCityStateZipInput.IsLive.ToString()}");
            Console.WriteLine($"License Key: {validateCityStateZipInput.LicenseKey}");


            CSZResponse response = ValidateCityStateZipClient.Invoke(validateCityStateZipInput);
            if (response.Error is null)
            {
                Console.WriteLine("\r\n* Validation *\r\n");

                Console.WriteLine($"City                          : {response.CityStateZip.City}");
                Console.WriteLine($"State                         : {response.CityStateZip.State}");
                Console.WriteLine($"ZIP Code                      : {response.CityStateZip.Zip}");
                Console.WriteLine($"General Delivery Service      : {response.CityStateZip.GeneralDeliveryService}");
                Console.WriteLine($"Street Service                : {response.CityStateZip.StreetService}");
                Console.WriteLine($"PO Box Service                : {response.CityStateZip.POBoxService}");
                Console.WriteLine($"PO Box Range Low              : {response.CityStateZip.POBoxRangeLow}");
                Console.WriteLine($"PO Box Range High             : {response.CityStateZip.POBoxRangeHigh}");
                Console.WriteLine($"Rural Route / Highway Contract: {response.CityStateZip.RRHCService}");
                Console.WriteLine($"Urbanization Service          : {response.CityStateZip.UrbanizationService}");
                Console.WriteLine($"Is Unique ZIP Code            : {response.CityStateZip.IsUniqueZipCode}");
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");

                Console.WriteLine($"Error Type     : {response.Error.Type}");
                Console.WriteLine($"Error Type Code: {response.Error.TypeCode}");
                Console.WriteLine($"Error Desc     : {response.Error.Desc}");
                Console.WriteLine($"Error Desc Code: {response.Error.DescCode}");
            }
        }
    }
}

File: address-validation-us-3-python-examples/validate_city_state_zip_rest_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../address-validation-us-3-python/REST"))

from validate_city_state_zip_rest import validate_city_state_zip

def validate_city_state_zip_rest_sdk_go(license, is_live_key):
    print("\n" + "-" * 57)
    print("Address Validation US 3 - ValidateCityStateZip - REST SDK")
    print("-" * 57)

    print("\n* Input *\n")
    print(f"City       : Santa Barbara")
    print(f"State      : CA")
    print(f"ZIP Code   : 93101")
    print(f"Is Live    : {is_live_key}")
    print(f"License Key: {license}")

    try:
        response = validate_city_state_zip(
            city="Santa Barbara",
            state="CA",
            zip="93101",
            license_key=license, 
            is_live=is_live_key
        )

        if not response.get('Error'):
            print("\n* Validation *\n")
            city_state_zip = response.get('CityStateZip', {})
            if city_state_zip:
                print(f"City                  : {city_state_zip.get('City', '')}")
                print(f"State                 : {city_state_zip.get('State', '')}")
                print(f"ZIP                   : {city_state_zip.get('Zip', '')}")
                print(f"GeneralDeliveryService: {city_state_zip.get('GeneralDeliveryService', '')}")
                print(f"POBoxService          : {city_state_zip.get('POBoxService', '')}")
                print(f"StreetService         : {city_state_zip.get('StreetService', '')}")
                print(f"RRHCService           : {city_state_zip.get('RRHCService', '')}")
                print(f"UrbanizationService   : {city_state_zip.get('UrbanizationService', '')}")
                print(f"POBoxRangeLow         : {city_state_zip.get('POBoxRangeLow', '')}")
                print(f"POBoxRangeHigh        : {city_state_zip.get('POBoxRangeHigh', '')}")
                print(f"IsUniqueZipCode       : {city_state_zip.get('IsUniqueZipCode', '')}")
            else:
                print("\n* Error *\n")
                err = response['Error']
                print(f"Error Type     : {err.get('Type', '')}")
                print(f"Error Type Code: {err.get('TypeCode', '')}")
                print(f"Error Desc     : {err.get('Desc', '')}")
                print(f"Error Desc Code: {err.get('DescCode', '')}")
    except RuntimeError as e:
        print("Error calling service:", e)
